src/main/scala/Configs.scala

```diff
@@ -34,15 +34,18 @@ class BaseConfig extends Config (
       entries += AddrMapEntry("bootrom", MemSize(4096, MemAttr(AddrMapProt.RX)))
       entries += AddrMapEntry("plic", MemRange(0x40000000, 0x4000000, MemAttr(AddrMapProt.RW)))
       entries += AddrMapEntry("prci", MemSize(0x4000000, MemAttr(AddrMapProt.RW)))
-      entries ++= site(ExtraDevices).map(_.addrMapEntry)
       new AddrMap(entries)
     }
     lazy val globalAddrMap = {
       val memBase = 0x80000000L
       val memSize = 0x10000000L
-      val io = new AddrMap(AddrMapEntry("int", internalIOAddrMap) +: site(ExtMMIOPorts).entries)
+
+      val intern = AddrMapEntry("int", internalIOAddrMap)
+      val extern = AddrMapEntry("ext", MemRange(0x50000000L, 0x30000000L, MemAttr(AddrMapProt.RWX)))
```

Comment:
Also, I think not having separate address maps would in fact be better, since it would allow finer-grain permissions in the Tile's TLB.

What if we added an optional boolean argument to the AddrMap like "collapse". Then we make the interconnect generator not recurse into it (treat it like a MemRange) if it's used as a submap.